### ISSUE TITLE
Decorators inside classes are falsely linted

### ISSUE THREAD

**Issue by UtkarshVerma** [microsoft/pyright#4505]
**Describe the bug**
Pyright lints my working code as erroneous. My code just utilizes decorators inside classes. 

**To Reproduce**
1. Type and save the code provided below.
2. Pyright will raise errors

**Expected behavior**
The code is perfectly fine and even gives the following output:
```
$ python test.py
Inside wrapper!
Hello, world!
```

**Screenshots or Code**
```python
class Test():

    def __init__(self) -> None:
        print(self.wrapped())

    def _decorate(func):

        def wrapper(self, *args, **kwargs):
            print("Inside wrapper!")
            return func(self, *args, **kwargs)

        return wrapper

    @_decorate
    def wrapped(self):
        return "Hello, world!"


Test()
```

Here are the errors it gives:
```
pyright 1.1.290
/home/subaru/test.py
  /home/subaru/test.py:14:6 - error: Argument of type "(self: Self@Test) -> Literal['Hello, world!']" cannot be assigned to parameter "func" of type "Self@Test" in function "_decorate"
    "object" is incompatible with "Test" (reportGeneralTypeIssues)
  /home/subaru/test.py:6:19 - warning: Instance methods should take a "self" parameter (reportSelfClsParameterName)
  /home/subaru/test.py:10:20 - error: Object of type "Test" is not callable (reportGeneralTypeIssues)
2 errors, 1 warning, 0 informations
Completed in 1.027sec
```

![image](https://user-images.githubusercontent.com/31820255/213899041-e41215c8-42d9-48b3-a1ba-69186deceba9.png)

**Comment by erictraut** [comment]
You need to mark the `_decorate` function as a `@staticmethod`. Otherwise it is interpreted an instance method with the first argument being `self`.

**Comment by UtkarshVerma** [comment]
Thanks!
